apps/liteprotocoltester/lightpush_publisher.nim

```diff
@@ -0,0 +1,110 @@
+import
+  std/strformat,
+  system/ansi_c,
+  chronicles,
+  chronos,
+  stew/byteutils,
+  stew/results,
+  json_serialization as js
+import
+  ../../../waku/common/logging,
+  ../../../waku/waku_node,
+  ../../../waku/node/peer_manager,
+  ../../../waku/waku_core,
+  ../../../waku/waku_lightpush/client,
+  ./tester_config,
+  ./tester_message
+
+proc prepareMessage(
+    sender: string,
+    messageIndex, numMessages: uint32,
+    startedAt: TimeStamp,
+    prevMessageAt: Timestamp,
+    contentTopic: ContentTopic,
+): WakuMessage =
+  let payload = ProtocolTesterMessage(
+    sender: sender,
+    index: messageIndex,
+    count: numMessages,
+    startedAt: startedAt,
+    sinceStart: getNowInNanosecondTime() - startedAt,
+    sincePrev: getNowInNanosecondTime() - prevMessageAt,
+  )
+
+  let text = js.Json.encode(payload)
+  let message = WakuMessage(
+    payload: toBytes(text), # content of the message
+    contentTopic: contentTopic, # content topic to publish to
+    ephemeral: true, # tell store nodes to not store it
+    timestamp: getNowInNanosecondTime(), # current timestamp
+  )
+
+  return message
+
+proc publishMessages(
+    wakuNode: WakuNode,
+    lightpushPubsubTopic: PubsubTopic,
+    lightpushContentTopic: ContentTopic,
+    numMessages: uint32,
+    delayMessages: Duration,
+) {.async.} =
+  let startedAt = getNowInNanosecondTime()
+  var prevMessageAt = startedAt
+  var failedToSendCount: uint32 = 0
+
+  let selfPeerId = $wakuNode.switch.peerInfo.peerId
+
+  var messagesSent: uint32 = 1
+  while numMessages >= messagesSent:
+    let message = prepareMessage(
+      selfPeerId, messagesSent, numMessages, startedAt, prevMessageAt,
+      lightpushContentTopic,
+    )
+    prevMessageAt = getNowInNanosecondTime()
+    let wlpRes = await wakuNode.lightpushPublish(some(lightpushPubsubTopic), message)
+
+    if wlpRes.isOk():
+      info "published message using lightpush",
+        index = messagesSent, count = numMessages
+    else:
+      error "failed to publish message using lightpush", err = wlpRes.error()
+      inc(failedToSendCount)
+
+    await sleepAsync(delayMessages) # Publish every 5 seconds
+    inc(messagesSent)
+
+  let report = catch:
+    """*----------------------------------------*
+|  Expected  |    Sent    |   Failed   |
+|{numMessages:>11} |{messagesSent-failedToSendCount-1:>11} |{failedToSendCount:>11} |
```

Comment:
This is great indeed!
In the future, I was envisioning to have a CSV with the following columns:
- use-case: e.g. 1000msg-1sender-1receiver; 1000-big-msgs-1-sender-1-receiver; etc.
- nwaku-version
- expected
- sent
- failed
- avg-latency
- ...

Reply:
Yes, it's a good idea! What about into phase2?
Indeed we will need to move to static sharding and variable length messages. Later on randomize used shards for auto sharding.
This table is intended to be human readable. For collecting and analysis I added the json formatted.

Reply:
> Yes, it's a good idea! What about into phase2? Indeed we will need to move to static sharding and variable length messages. Later on randomize used shards for auto sharding. This table is intended to be human readable. For collecting and analysis I added the json formatted.

Cool! Sure, let's do it in further PRs :)